src/material-examples/select-error-state-matcher/select-erorr-state-matcher-example.html

```diff
@@ -0,0 +1,12 @@
+<mat-form-field>
+  <mat-select placeholder="Choose one" [formControl]="selected" [errorStateMatcher]="matcher">
+    <mat-option>Clear</mat-option>
+    <mat-option value="valid">Valid option</mat-option>
+    <mat-option value="invalid">Invalid option</mat-option>
+  </mat-select>
+  <mat-hint>Errors appear instantly!</mat-hint>
+  <mat-error *ngIf="selected.hasError('required')">You must make a selection</mat-error>
+  <mat-error *ngIf="selected.hasError('pattern') && !selected.hasError('required')">
```

Comment:
I'm not sure how common of a use case it is to have pattern validation on a select since the developer provides the possible values beforehand.

Reply:
Yeah its not the most practical example, but I just wanted to show a couple different error states